validation/telemetry/event.4.headless-sample.fail.json

```diff
@@ -0,0 +1,382 @@
+  "creationDate": "2018-05-24T16:59:14.497Z",
+  "version": 4,
+  "application": {
+    "architecture": "x86-64",
+    "buildId": "20180509105613",
+    "name": "Firefox",
+    "version": "62.0a1",
+    "displayVersion": "62.0a1",
+    "vendor": "Mozilla",
+    "platformVersion": "62.0a1",
+    "xpcomAbi": "x86_64-gcc3",
+    "channel": "default"
+  },
+  "payload": {
+    "reason": "shutdown",
+    "processStartTimestamp": 1527179580000,
+    "sessionId": "23929b57-04c3-4a6d-bcb6-150041eb4c54",
+    "subsessionId": "cf96de8a-9307-49dc-9f52-b3b834296d68",
+    "lostEventsCount": 0,
+    "events": {
+      "parent": [
+        [
+          1572303,
+          "devtools.main",
+          "close",
+          "tools",
+          null,
+          {
+            "host": "bottom",
+            "width": "1300"
+          }
+        ],
+        [
+          1572307,
+          "devtools.main",
+          "exit",
+          "webconsole",
+          null,
+          {
+            "host": "bottom",
+            "width": "1300",
+            "panel_name": "webconsole",
+            "next_panel": "none",
+            "reason": "toolbox_close"
+          }
+        ]
+      ]
+    }
+  },
+  "clientId": "ac69287d-2ebb-4cba-b3ec-558a677d0a5e",
+  "environment": {
+    "build": {
+      "applicationId": "{ec8030f7-c20a-464f-9b0e-13a3a9e97384}",
+      "applicationName": "Firefox",
+      "architecture": "x86-64",
+      "buildId": "20180509105613",
+      "version": "62.0a1",
+      "vendor": "Mozilla",
+      "displayVersion": "62.0a1",
+      "platformVersion": "62.0a1",
+      "xpcomAbi": "x86_64-gcc3",
+      "updaterAvailable": true
+    },
+    "partner": {
+      "distributionId": null,
+      "distributionVersion": null,
+      "partnerId": null,
+      "distributor": null,
+      "distributorChannel": null,
+      "partnerNames": []
+    },
+    "system": {
+      "memoryMB": 32114,
+      "virtualMaxMB": null,
+      "cpu": {
+        "count": 8,
+        "cores": 4,
+        "vendor": "GenuineIntel",
+        "family": 6,
+        "model": 60,
+        "stepping": 3,
+        "l2cacheKB": 256,
+        "l3cacheKB": 8192,
+        "speedMHz": 4000,
+        "extensions": [
+          "hasMMX",
+          "hasSSE",
+          "hasSSE2",
+          "hasSSE3",
+          "hasSSSE3",
+          "hasSSE4_1",
+          "hasSSE4_2",
+          "hasAVX",
+          "hasAVX2",
+          "hasAES"
+        ]
+      },
+      "os": {
+        "name": "Linux",
+        "version": "4.4.0-112-generic",
+        "locale": "en-CA"
+      },
+      "hdd": {
+        "profile": {
+          "model": null,
+          "revision": null
+        },
+        "binary": {
+          "model": null,
+          "revision": null
+        },
+        "system": {
+          "model": null,
+          "revision": null
+        }
+      },
+      "gfx": {
+        "D2DEnabled": null,
+        "DWriteEnabled": null,
+        "ContentBackend": "Skia",
+        "Headless": 0,
```

Comment:
Could you pare down this validation ping? One thought is to use the `main.4.min.pass` document as a base to avoid the extra boilerplate.

https://github.com/mozilla-services/mozilla-pipeline-schemas/blob/151c1dbbac66f7c52285e32aa7edc494957e664a/validation/telemetry/main.4.min.pass.json#L1-L16

The cases themselves look good otherwise.

Reply:
Alrighty. It took a little finagling since introducing the optional `environment` means you need to add some required things (like almost the entirety of `environment.build`), but I think this is now the minimal example of a passing- and failing-due-to-`Headless` schema validation set.